services.py

```diff
@@ -104,3 +107,72 @@ def get_all_services(self):
                     .all())
         return services
 
+class InvoiceService:
+    
+    @classmethod
+    def get_invoice_payment_log(self):
+        '''Get the payment info for all invoices'''
+        invoice_payment_info = []
+        invoices = Invoice.query.all()
+        payments = Payment.query.all()
```

Comment:
This could be very expensive. Imagine having millions of invoices and payments and having to pull them into memory every time. You should have invoice.payments already if the relationship is declared.